[Feature] Inserting Notes Functionality

Civolilah opened the pull request:
In progress...

turbo124 commented:
@Civolilah client overview -click on activitiers table => white page

Civolilah commented:
> @Civolilah client overview -click on activitiers table => white page

Civolilah commented:
> @Civolilah client overview -click on activitiers table => white page

@turbo124 Hmm, I just did a fresh seed of the data, and the `cypress` client does not have any activities, but on the 'Activity' tab, I did not get a white page.

![Screenshot 2024-07-17 at 05 53 22](https://github.com/user-attachments/assets/90bc4723-5027-453c-a1ee-c1e405344cb0)

Also, I've created new client and I had the activity for creation that one client.

![Screenshot 2024-07-17 at 05 51 24](https://github.com/user-attachments/assets/69942e1a-0089-4749-9e25-b41b29cb6d0a)

I've added "optional chaining" in one case to avoid all possibilities of getting a white page, but I do not think that was the reason for the white page. I've kept everything the same except for the implementation of the "plus icon" with the modal and insert notes activity.

Can you please let me know if you get the white page on the develop branch in the same case as on this branch, or if you are aware of the exact case? It would be helpful for me.

turbo124 commented:
@Civolilah i think the issue relates to missing .label prop?

![image](https://github.com/user-attachments/assets/20505274-948b-4277-950b-1acd2a273bbb)

Civolilah commented:
> @Civolilah i think the issue relates to missing .label prop?
> 
> ![image](https://private-user-images.githubusercontent.com/5827962/349706819-20505274-948b-4277-950b-1acd2a273bbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjM4MjcsIm5iZiI6MTcyMTI2MzUyNywicGF0aCI6Ii81ODI3OTYyLzM0OTcwNjgxOS0yMDUwNTI3NC05NDhiLTQyNzctOTUwYi0xYWNkMmEyNzNiYmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDA0NTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjM5YjYyYjBkNTM3ZWQ0ZjA4OGY5NzZkZWQyNjQ4ZTA2MjZkZGM3YmRlOTJlYzZmMDgzNjZlMjEwNTIxYjY5NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.G-80zW2NBlLGRnVPJW8r96jx0JMyJWGoEFx-htAMyn8)

@turbo124 Ah, okay, thanks. I just added the fix for this issue. Please let me know if it works correctly on your end now.

turbo124 commented:
@Civolilah we have a follow up task in Jira for this, i'll tag this as pending revision until completed.

Civolilah commented:
> @Civolilah we have a follow up task in Jira for this, i'll tag this as pending revision until completed.

@turbo124 Hmm, I researched all current Jira tasks, but I could not find that ticket/Jam. Can you please just help me find the correct ticket?

Civolilah commented:
@turbo124 As we discussed on Slack, the implementation for the functionality has slightly changed with the addition of the `comments_only` query parameter. Adding comments is now implemented on the client show page, in the more_actions dropdown across entities where it is needed, and also in sliders for entities that have it. Screenshots of the implementation:

<img width="369" alt="Screenshot 2024-07-22 at 10 39 08" src="https://github.com/user-attachments/assets/fd83d8e5-87b8-47a4-b251-68fcfe7aa6bc">

<img width="1050" alt="Screenshot 2024-07-22 at 10 39 26" src="https://github.com/user-attachments/assets/8f2192e8-dc64-4f16-8820-8b52a8ee624f">

<img width="240" alt="Screenshot 2024-07-22 at 10 39 38" src="https://github.com/user-attachments/assets/5befbae4-0c03-4d7a-bf41-1b68bf492ab2">

Let me know your thoughts.

turbo124 commented:
![image](https://github.com/user-attachments/assets/8c975b5b-f3d1-4800-a343-602ee7461383)


Seeing this when viewing Client Activity from Overview

Civolilah commented:
> ![image](https://private-user-images.githubusercontent.com/5827962/351616281-8c975b5b-f3d1-4800-a343-602ee7461383.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MTQ2NTEsIm5iZiI6MTcyMTgxNDM1MSwicGF0aCI6Ii81ODI3OTYyLzM1MTYxNjI4MS04Yzk3NWI1Yi1mM2QxLTQ4MDAtYTM0My02MDJlZTc0NjEzODMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMDk0NTUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzZhOTExY2YzMzI0MWQ4MTJiMzUzOTQwYjFlYzcxMjcwNTEyZjgwZTk5ODlmNDJlYmNkYThhZGFiNDNlNTVmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.vZ7oK-688WioJBWXqn-gsqR2h5YcqfDPJ6ohwbUMa1s)
> 
> Seeing this when viewing Client Activity from Overview

@turbo124 I made changes to fix it. Please let me know if it is fixed on your end as well.

turbo124 commented:
@Civolilah should the comments toggle work on client overview?

Civolilah commented:
> @Civolilah should the comments toggle work on client overview? (ie, only comments are shown)
> 
> also, when inserting a note, the activity list does not update to show the new comment

@turbo124 On my end, the toggle correctly sets the `comment_only` query param to `true`. Should it be the expected behavior? Screenshot:

<img width="525" alt="Screenshot 2024-07-25 at 10 00 57" src="https://github.com/user-attachments/assets/f033b368-d09b-4d7a-8ee1-3745f4485d77">

Also, I've implemented the correct invalidation of activities queries, so now the list should update correctly everywhere after adding a comment.

Let me know your thoughts.

turbo124 commented:
@Civolilah it is not working for me, when i use the toggle, nothing changes in the UI, the activities + comments still appears all together.

Civolilah commented:
> @Civolilah it is not working for me, when i use the toggle, nothing changes in the UI, the activities + comments still appears all together.

@turbo124 As we discussed on Slack, the filtering has been done on the React side, so the toggle should work as expected right now. Let me know your thoughts.

turbo124 commented:
@Civolilah thanks this looks good.

How are we handling line breaks? are these being passed through as \n ? We'll want to ensure they are preserved and also displayed as expected.

Civolilah commented:
> @Civolilah thanks this looks good.
> 
> How are we handling line breaks? are these being passed through as \n ? We'll want to ensure they are preserved and also displayed as expected.

@turbo124 Yes, the line breaks are passed through as `\n`, and I've added the implementation for it so that every line break is displayed as expected. Screenshot:

![Screenshot 2024-07-28 at 19 02 38](https://github.com/user-attachments/assets/4bdcfe64-939a-41c5-8df8-0c8eb26b454d)

`Important note`: We had notes displayed on the right side of the IP address as an additional place. I believe that is unnecessarily special, especially if we have line breaks in entered notes, as it slightly breaks the UI. So basically, I just removed the display of notes on the right side of the IP address and displayed them only where we implemented it in this PR.

Let me know your thoughts.

turbo124 commented:
I think to notes are tough to read when arranging them in a column. I think it works better by adding them in as a row / after a line break if possible?

Civolilah commented:
> I think to notes are tough to read when arranging them in a column. I think it works better by adding them in as a row / after a line break if possible?

@turbo124 The UI for displaying notes has been slightly changed, as we discussed on Slack. The notes will now be in a new row, separated from the rest of the text above. Let me now your thoughts.
